app.js

```diff
@@ -1,21 +1,31 @@
 const { getTopics } = require("./controllers/topics.controllers")
 const { getEndpoints } = require("./controllers/api.controllers")
+const { getArticleById } = require("./controllers/articles.controllers")
 
 const express = require("express")
 
 const app = express()
 
-app.use(express.json())
-
 app.get("/api/topics", getTopics)
 
 app.get("/api", getEndpoints)
 
+app.get("/api/articles/:article_id", getArticleById)
+
+app.all('*', (req, res) => {
+    res.status(404).send({msg: "The request path does not exist"})
+})
+
 app.use((err, req, res, next) => {
+    if (err.msg) {
+        res.status(err.status).send({msg: err.msg})
```

Comment:
Nice custom error handling middleware. You could always look at refactoring these functions into another file like we have done with the controllers just to keep this file a bit smaller but that's just a suggestion 😊